lib/rules/eqeqeq.js

```diff
@@ -20,13 +20,30 @@ module.exports = {
         schema: [
             {
                 enum: ["always", "smart", "allow-null"]
+            },
+            {
+                type: "object",
+                properties: {
+                    null: {
+                        enum: ["always", "never", "ignore"]
+                    }
+                },
+                additionalProperties: false
             }
```

Comment:
This seems reasonable. But I'm not too sure about JSON schema so I will leave it as is unless you feel strongly about dropping in the above code.